File: weather_fetcher.js
// Weather Fetcher in JavaScript (Node.js)
const fetch = require('node-fetch');
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function getWeather(city, apiKey) {
  const url = `https://api.openweathermap.org/data/2.5/weather?q=${encodeURIComponent(city)}&appid=${apiKey}&units=metric`;
  fetch(url)
    .then(res => res.json())
    .then(data => {
      if (data.cod === 200) {
        console.log(`\nWeather in ${city}:`);
        console.log(`Temperature: ${data.main.temp}°C`);
        console.log(`Feels like: ${data.main.feels_like}°C`);
        console.log(`Condition: ${data.weather[0].description}`);
      } else {
        console.log(`\n[Error] ${data.message}`);
      }
      rl.close();
    })
    .catch(err => {
      console.log('\n[Error] Could not fetch weather data.');
      rl.close();
    });
}

console.log('=== Weather Fetcher (OpenWeatherMap) ===');
rl.question('Enter your OpenWeatherMap API key: ', apiKey => {
  rl.question('Enter city name: ', city => {
    getWeather(city, apiKey.trim());
  });
});

File: weather_fetcher.py
import requests

def get_weather(city, api_key):
    url = f"https://api.openweathermap.org/data/2.5/weather?q={city}&appid={api_key}&units=metric"
    response = requests.get(url)
    if response.status_code == 200:
        data = response.json()
        main = data.get('main', {})
        weather = data.get('weather', [{}])[0]
        print(f"\nWeather in {city}:")
        print(f"Temperature: {main.get('temp', 'N/A')}°C")
        print(f"Feels like: {main.get('feels_like', 'N/A')}°C")
        print(f"Condition: {weather.get('description', 'N/A').title()}")
    else:
        print("\n[Error] Could not fetch weather data. Please check the city name or your API key.")

if __name__ == "__main__":
    print("=== Weather Fetcher (OpenWeatherMap) ===")
    api_key = input("Enter your OpenWeatherMap API key: ").strip()
    city = input("Enter city name: ").strip()
    get_weather(city, api_key)
